prog_question_test/tests/autograde.py

```diff
@@ -41,6 +41,28 @@ def test_private_2(self):
 
         self.assertTrue(test_result('private', 2))
 
+    @timeout_decorator.timeout(2)
+    def test_recursion(self):
+        def is_recursion(func):
+            def is_target(node, name):
+                return type(node) == ast.Call and type(node.func) == ast.Name and node.func.id == name
+
+            def is_recursion_node(func_def_node):
+                print(func_def_node)
+                name = func_def_node.name
+                return len(list(filter(lambda node: is_target(node, name), ast.walk(func_def_node)))) > 0
+
+            root_node = ast.parse(inspect.getsource(func)).body[0]
+            func_def_nodes = filter(lambda node: type(node) == ast.FunctionDef, ast.walk(root_node))
+            return len(list(filter(lambda node: is_recursion_node(node), func_def_nodes))) > 0
+
+        self.meta['expression'] = 'test recursion'
+        self.meta['expected'] = 'True'
+        self.meta['output'] = is_recursion(user_function)

```

Comment:
should be `str(is_recursion(user_function))` because this is the printout of the result. Not sure if there will be implicit conversions though...